src/main/java/org/researchspace/services/storage/s3/S3Storage.java

```diff
@@ -0,0 +1,270 @@
+
+            com.amazonaws.services.s3.model.ObjectMetadata s3Metadata = s3.getObjectMetadata(config.getBucket(), key);
+
+            // If metadata exist
+            String user = s3Metadata.getUserMetaDataOf("Author");
+            S3StorageLocation location = new S3StorageLocation(config.getBucket(), key);
+            ObjectMetadata metadata = new ObjectMetadata(Objects.isNull(user) ? "" : user,
+                    s3Metadata.getLastModified().toInstant());
+
+            return Optional.of(new ObjectRecord(location, path, revision, metadata));
+        } catch (AmazonS3Exception e) {
+            // If the key is missing, return an empty result
+            if (StringUtils.equals(e.getErrorCode(), "NoSuchKey")
+                    || StringUtils.equals(e.getErrorCode(), "404 Not Found"))
+                return Optional.empty();
+            throw new StorageException(e.getMessage());
+        } catch (Exception e) {
+            throw new StorageException(e.getMessage());
+        }
+
+    }
+
+    @Override
+    public List<ObjectRecord> getRevisions(StoragePath path) throws StorageException {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public List<ObjectRecord> getAllObjects(StoragePath prefix) throws StorageException {
+
+        Optional<StoragePath> mappedPrefix = paths.mapForward(prefix);
+
+        if (!mappedPrefix.isPresent()) {
+            throw new StorageException("");
+        }
+
+        String pref = mappedPrefix.get().toString();
+
+        try {
+
+            ListObjectsV2Result result = s3.listObjectsV2(config.getBucket(), pref);
+            List<ObjectRecord> records = new ArrayList<>();
+
+            List<S3ObjectSummary> objects = result.getObjectSummaries();
+            for (S3ObjectSummary os : objects) {
+
+                Optional<StoragePath> path = StoragePath.tryParse(os.getKey()).flatMap(paths::mapBack);
+
+                if (path.isPresent()) {
+                    Optional<ObjectRecord> objectRecord = getObject(path.get(), null);
+                    if (objectRecord.isPresent()) {
+                        records.add(objectRecord.get());
+                    }
+                }
+
+            }
+
+            return records;
+
+        } catch (Exception e) {
+
+            throw new StorageException(e.getMessage());
+        }
+    }
+
+    @Override
+    public ObjectRecord appendObject(StoragePath path, ObjectMetadata metadata, InputStream content, long contentLength)
+            throws StorageException {
+
+        try {
+            ObjectRecord objectRecord;
+
+            String key = getKeyFromStoragePath(path);
+
+            S3StorageLocation location = new S3StorageLocation(config.getBucket(), key);
+
+            // TODO: calculate revision
+            objectRecord = new ObjectRecord(location, path, "", metadata);
+
+            // TODO: check PutObjectResult
+            s3.putObject(config.getBucket(), key, content.toString());
```

Comment:
It doesn't work, you probably need to use other method that takes input stream and metadata - https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Client.html#putObject-java.lang.String-java.lang.String-java.io.InputStream-com.amazonaws.services.s3.model.ObjectMetadata-

Reply:
Fix in #5fa15d7